src/solvers/flattening/arrays.cpp

```diff
@@ -49,11 +48,13 @@ literalt arrayst::record_array_equality(
   // check types
   if(!base_type_eq(op0.type(), op1.type(), ns))
   {
-    std::cout << equality.pretty() << '\n';
-    throw "record_array_equality got equality without matching types";
+    prop.error() << equality.pretty() << messaget::eom;
+    DATA_INVARIANT(false, "record_array_equality got equality without matching types");
   }
 
-  assert(ns.follow(op0.type()).id()==ID_array);
+  DATA_INVARIANT(
+    ns.follow(op0.type()).id()==ID_array,
+    "record_array_equality parameter should be array-typed");
```

Comment:
This is a lovely example of how to convert asserts and how to add meaningful information during the conversion.